plugins/kubernetes-backend/src/service/KubernetesProxy.ts

```diff
@@ -0,0 +1,286 @@
+      );
+    }
+    return {};
+  }
+
+  protected async getClusterDetails(
+    clusterSupplier: KubernetesClustersSupplier,
+    krc: KubernetesProxyClusters,
+  ): Promise<ClusterDetails[]> {
+    const clusters = await clusterSupplier.getClusters();
+
+    const clusterNames = Object.keys(krc);
+
+    const clusterDetails = clusters.filter(c => clusterNames.includes(c.name));
+
+    const clusterDetailsAuth = await Promise.all(
+      clusterDetails.map(async c => {
+        const cAuth: ClusterDetails = Object.assign(c, {
+          serviceAccountToken: krc[c.name],
+        });
+        return cAuth;
+      }),
+    );
+
+    return clusterDetailsAuth;
+  }
+
+  protected getClusterURI(details: ClusterDetails): string {
+    const client = this.getKubeConfig(details);
+    return client.getCurrentCluster()?.server || '';
+  }
+
+  protected async makeRequestToCluster(
+    details: ClusterDetails,
+    req: Request,
+  ): Promise<KubernetesProxyResponse> {
+    const serverIP = this.getClusterURI(details);
+    if (!serverIP) {
+      this.logger.error(
+        `could not find server IP for cluster with name ${details.name}`,
+      );
+      return {
+        code: ERROR_INTERNAL_SERVER,
+        data: null,
+      };
+    }
+
+    const query = decodeURIComponent(req.params.encodedQuery) || '';
+    const uri = `${serverIP}/${query}`;
+
+    const contentType = req.header(HEADER_CONTENT_TYPE) || APPLICATION_JSON;
+
+    const res = await this.sendClusterRequest(
+      details,
+      uri,
+      req.method,
+      contentType,
+      req.body,
+    );
+
+    return res;
+  }
+
+  protected async sendClusterRequest(
+    details: ClusterDetails,
+    uri: string,
+    method: string,
+    contentType: string,
+    body?: any,
+  ): Promise<KubernetesProxyResponse> {
+    const bearerToken = details.serviceAccountToken;
+    if (!bearerToken) {
+      this.logger.error('could not find bearer token!');
+      return {
+        code: ERROR_BAD_REQUEST,
+        data: {
+          error: 'Invalid service account token',
+        },
+      };
+    }
+
+    const reqData: any = {
+      method,
+      headers: {
+        'Content-Type': contentType,
+        Authorization: `Bearer ${bearerToken}`,
+      },
+    };
+
+    if (!details.skipTLSVerify) {
+      if (details.caData) {
+        const ca = bufferFromFileOrString('', details.caData)?.toString() || '';
+        reqData.agent = new https.Agent({ ca });
+      } else {
+        this.logger.info('could not find CA certificate!');
+        return {
+          code: ERROR_INTERNAL_SERVER,
+          data: {
+            error: 'Invalid CA certificate configured within Backstage',
+          },
+        };
+      }
+    }
+
+    if (body && Object.keys(body).length > 0) {
+      reqData.body = JSON.stringify(body);
+    }
+
+    try {
+      const req = await fetch(uri, reqData);
+
+      let res;
+      if (contentType === APPLICATION_JSON) {
```

Comment:
Could need to be a bit more lax here right ... some respond with a `; charset=utf8` and similar. So probably should be `contentType === APPLICATION_JSON || contentType.startsWith(APPLICATION_JSON)` or something along those lines